ACE/ace/Condition_Recursive_Thread_Mutex.h

```diff
@@ -101,11 +108,22 @@ class ACE_Export ACE_Condition<ACE_Recursive_Thread_Mutex>
   ACE_Recursive_Thread_Mutex &mutex_;
 
 };
+// *NOTE*: prevent implicit instantiations by includees to relieve the linker
+#if defined (__GNUG__)
```

Comment:
Shouldn't this have a version check on g++ version, maybe add a new ACE_HAS... macro because probably this is then needed on more places?